src/evaluator.ts

```diff
@@ -154,27 +234,61 @@ export class Evaluator {
       }
 
       // evaluation succeded, find variation in flag
-      return this.findVariation(fc, identifier);
+      return identifier;
     }
     // there is no rules return undefined
     return undefined;
   }
 
-  private evaluateFlag(fc: FeatureConfig, target: Target): Variation | undefined {
-    if (fc.state === FeatureState.Off) {
-      return this.findVariation(fc, fc.offVariation);
-    }
-    let variation = this.evaluateRules(fc, target);
-    if (!variation) {
-      // if variation is undefined then use flag default serve
-      if (fc.defaultServe.distribution) {
-        const identifier = this.getKeyName(fc.defaultServe.distribution, target);
-        variation = this.findVariation(fc, identifier);
-      } else {
-        variation = this.findVariation(fc, fc.defaultServe.variation);
+  private evaluateVariationMap(
+    variationToTargetMap: VariationMap[],
+    target: Target,
+  ): string | undefined {
+    if (!variationToTargetMap) return undefined;
+
+    for (const variationMap of variationToTargetMap) {
+      // find target
+      const targetMap = variationMap.targets?.find(
+        (elem) => elem.identifier === target.identifier,
+      );
+      if (targetMap) return variationMap.variation;
+
+      // find target in segment
+      const segmentIdentifiers = variationMap.targetSegments;
+      if (
+        segmentIdentifiers &&
+        this.isTargetIncludedOrExcludedInSegment(segmentIdentifiers, target)
+      ) {
+        return variationMap.variation;
       }
     }
-    return variation
+
+    return undefined;
+  }
+
+  private evaluateFlag(
+    fc: FeatureConfig,
+    target: Target,
+  ): Variation | undefined {
+    if (fc.state === FeatureState.Off) {
+      return this.findVariation(fc.variations, fc.offVariation);
+    }
+
+    // process first variationMaps
+    let variation = this.evaluateVariationMap(fc.variationToTargetMap, target);
+    if (variation) return this.findVariation(fc.variations, variation);
+
+    // not found in variationMaps proceed with rules processor
+    variation = this.evaluateRules(fc.rules, target);
+    if (variation) return this.findVariation(fc.variations, variation);
+
+    // not found in rules proceed with default serve
+    variation = fc.defaultServe.variation;
+    // default serve variation is undefined so continue with distribution
```

Comment:
It doesn't appear that you are testing whether the default serve variation is undefined